### PR TITLE
Put metrics fields to bottom of EMF log entry

### PR DESCRIPTION
**Pull request by vastin**
**Tracking Issues:**

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26697

**Description:**
It is a short-term solution to workaround CW log insight limitation by placing metrics fields to the bottom of log entry.

**Testing:**
Passed unit test and benchmark result showed no noticeable difference.

### CODE REVIEW

**Review by mxiamxia — APPROVED** [review]
LGTM!

File: exporter/awsemfexporter/metric_translator_test.go [hunk 1]
```diff
@@ -405,7 +405,7 @@ func TestTranslateCWMetricToEMF(t *testing.T) {
...
+			expectedEMFLogEvent:     "{\"OTelLib\":\"cloudwatch-otel\",\"Sources\":[\"cadvisor\",\"pod\",\"calculated\"],\"Version\":\"1\",\"_aws\":{\"CloudWatchMetrics\":[{\"Namespace\":\"test-emf\",\"Dimensions\":[[\"OTelLib\"],[\"OTelLib\",\"spanName\"]],\"Metrics\":[{\"Name\":\"spanCounter\",\"Unit\":\"Count\"}]}],\"Timestamp\":1596151098037},\"kubernetes\":{\"container_name\":\"cloudwatch-agent\",\"docker\":{\"container_id\":\"fc1b0a4c3faaa1808e187486a3a90cbea883dccaf2e2c46d4069d663b032a1ca\"},\"host\":\"ip-192-168-58-245.ec2.internal\",\"labels\":{\"controller-revision-hash\":\"5bdbf497dc\",\"name\":\"cloudwatch-agent\",\"pod-template-generation\":\"1\"},\"namespace_name\":\"amazon-cloudwatch\",\"pod_id\":\"e23f3413-af2e-4a98-89e0-5df2251e7f05\",\"pod_name\":\"cloudwatch-agent-26bl6\",\"pod_owners\":[{\"owner_kind\":\"DaemonSet\",\"owner_name\":\"cloudwatch-agent\"}]},\"spanName\":\"test\",\"spanCounter\":0}",
```
Comment (lisguo) [inline exporter/awsemfexporter/metric_translator_test.go:408]: Is this test testing the above changes? I see that this is just the same thing but reversing spanName and spanCounter?

Comment (vastin) [reply exporter/awsemfexporter/metric_translator_test.go:408]: Yes, it is covered. This change will only move metrics (spanCounter in this case) to the bottom of JSON string. It shouldn't change any content in JSON.

Comment (lisguo) [reply exporter/awsemfexporter/metric_translator_test.go:408]: Gotcha, didn't realize spanCounter was the metric name here. LGTM

**Review by lisguo — APPROVED** [review]

**Pull request merged**